AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Subscribe/DataStoreObserveQueryOperation.swift

```diff
@@ -198,52 +214,23 @@ public class AWSDataStoreObserveQueryOperation<M: Model>: AsynchronousOperation,
             if self.observeQueryStarted.getAndSet(true) {
                 return
             }
+            if let storageEngine = storageEngine {
+                self.storageEngine = storageEngine
+            }
             self.log.verbose("Start ObserveQuery")
-            self.dataStoreEventSink = Amplify.Hub.publisher(for: .dataStore)
-                .sink(receiveValue: self.onDataStoreEvent(hubPayload:))
             self.subscribeToItemChanges()
-            self.setInitialIsSyncedState()
             self.initialQuery()
         }
     }
 
-    // MARK: - Initial IsSynced state
-
-    func setInitialIsSyncedState() {
-        // if the sync engine is active, then `.ready` event has already fired, set isSynced to true
-        if let storageAdapter = storageEngine as? StorageEngine,
-           let remoteSyncEngine = storageAdapter.syncEngine as? RemoteSyncEngine,
-           case .syncEngineActive = remoteSyncEngine.stateMachine.state {
-            isSynced.set(true)
-        }
-    }
-
-    // MARK: - Observe DataStore events
-
-    func onDataStoreEvent(hubPayload: HubPayload) {
-        if isCancelled || isFinished {
-            finish()
-            return
-        }
-
-        if hubPayload.eventName == HubPayload.EventName.DataStore.modelSynced && !isSynced.get(),
-           let modelSynced = hubPayload.data as? ModelSyncedEvent, modelSynced.modelName == modelSchema.name {
-            isSynced.set(true)
-            log.verbose("Received .modelSynced event")
-        } else if hubPayload.eventName == HubPayload.EventName.DataStore.ready && !isSynced.get() {
-            isSynced.set(true)
-            log.verbose("Received .ready event")
-        }
-    }
-
     // MARK: - Query
 
     func initialQuery() {
         if isCancelled || isFinished {
             finish()
             return
         }
-
+        log.verbose("Initial Query")
```

Comment:
nit: if `initialQuery` is called multiple times, we can remove the log statement